lua/plenary/path.lua

```diff
@@ -237,6 +235,23 @@ function Path:normalize(cwd)
   return self.filename
 end
 
+local function shorten_len(filename, len)
+  local final_match
+  local final_path_components = {}
+  for match in (filename..path.sep):gmatch("(.-)"..path.sep) do
+    if #match > len then
+      table.insert(final_path_components, #final_path_components, string.sub(match, 1, len))
```

Comment:
you can just do `table.insert(tbl, item)`. Don't calculate the length of the path.